File: kmeans/haskell/sequential/src/DistanceMetrics.hs
module DistanceMetrics where

squaredDiff :: Double -> Double -> Double
squaredDiff x y = (y - x)^2

euclideanDistance :: [Double] -> [Double] -> Double
euclideanDistance x y = sqrt c
    where c = foldl1 (+) (zipWith (squaredDiff) x y)

File: kmeans/charm++/sequential/src/distanceMetrics.cpp
#include <cmath>
#include <vector>
#include "distanceMetrics.hpp"

namespace DM {
    double DistanceMetrics::euclideanDistance(const std::vector<double>& vec1, const std::vector<double>& vec2) {
        int counter = 0;
        double sum_of_squares = 0;
        for(double point : vec1) {
            sum_of_squares += std::pow((point - vec2[counter]), 2);
            counter++;
        }
        return std::sqrt(sum_of_squares);
    }
}